### PR TITLE
Add MigrationChecker class and GoldBarController skeleton. Also remove AutoMigrationHandler.

### PR DESCRIPTION
**Pull request by bigfluffycookie**
part of https://github.com/SonarSource/sonarlint-visualstudio/issues/4254

### CODE REVIEW

File: src/ConnectedMode/Migration/IGoldBarController.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel.Composition;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Migration
+{
+    public interface IGoldBarController
```
Comment (bigfluffycookie) [inline src/ConnectedMode/Migration/IGoldBarController.cs:25]: Empty goldbar controller so I could properly test the MigrationChecker

File: src/ConnectedMode/Migration/MigrationChecker.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+            activeSolutionTracker.ActiveSolutionChanged += OnActiveSolutionChanged;
+
+            // Initial check incase event was fired before we registered.
+            DisplayGoldBarIfMigrationIsNeeded();
+        }
+
+        private void OnActiveSolutionChanged(object sender, ActiveSolutionChangedEventArgs args)
+        {
+            if (args.IsSolutionOpen)
```
Comment (bigfluffycookie) [inline src/ConnectedMode/Migration/MigrationChecker.cs:57]: Do we need to close the goldbar if the solution closes?

File: src/ConnectedMode.UnitTests/Migration/MigrationCheckerTests.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        public void MefCtor_CheckIsExported()
+        {
+            MefTestHelpers.CheckTypeCanBeImported<MigrationChecker, MigrationChecker>(
+                MefTestHelpers.CreateExport<IActiveSolutionTracker>(),
+                MefTestHelpers.CreateExport<IGoldBarController>(),
+                MefTestHelpers.CreateExport<IConfigurationProvider>(),
+                MefTestHelpers.CreateExport<IObsoleteConfigurationProvider>()); 
+        }
+
+        [TestMethod]
```
Comment (bigfluffycookie) [inline src/ConnectedMode.UnitTests/Migration/MigrationCheckerTests.cs:42]: This test is more or less the same as it was in the AutoMigrationHandlerTests.cs

File: src/ConnectedMode/Migration/MigrationChecker.cs (hunk 2 above)
Comment (duncanp-sonar) [reply src/ConnectedMode/Migration/MigrationChecker.cs:57]: Yes, I think we should.

File: src/ConnectedMode/Migration/IGoldBarController.cs (hunk 1 above)
Comment (duncanp-sonar) [reply src/ConnectedMode/Migration/IGoldBarController.cs:25]: 'IGoldBarController` is pretty generic. How about something more specific e.g. `IMigrationGoldBarController`.

Or perhaps something even more specific such as `IMigrationPrompt` with a method called `Show` - the fact that it is displaying a gold bar is an implementation detail from the point of view of the caller.

**Review by duncanp-sonar — APPROVED** [review]
Looks good

**Pull request merged**